### PR TITLE
Update projects on Commission update

### PR DESCRIPTION
**Pull request by richpryce**
## What does this change?
When a commission status is changed the status of associated projects changes according to these rules:

Commission “Completed” → all projects NOT Completed or Killed should be set to Completed
Commission “Held” → all projects NOT Completed or Killed or Held should be set to Held
Commission “In Progress” → no change
Commission "Killed" -> all projects NOT Completed or Killed should be set to Killed

##  Implementation
When a commission is updated the `CommissionStatusPropagator` is invoked. This calls `getProjectsEligibleForStatusChange` from `ProjectEntryController`. The returned projects status' are updated in the db and rabbitMQ messages are sent.

## How to test
Change the status of a commission in Pluto and see the associated projects update in line with the rules listed above.

### CODE REVIEW

File: app/controllers/ProjectEntryController.scala [hunk 1]
```diff
@@ -97,6 +97,13 @@ class ProjectEntryController @Inject() (@Named("project-creation-actor") project
     }
   }
 
+  def updateCommissionProjects(newStatus: EntryStatus.Value, commissionId: Int): Future[Seq[Try[Int]]] = {
+    val action: DBIO[Seq[(Int, ProjectEntry)]] = ProjectEntry.dbActionForStatusUpdate(newStatus, commissionId)
+    dbConfig.db.run(action).flatMap { projectTuples =>
```
Comment (paulmr) [inline app/controllers/ProjectEntryController.scala:102]: possible that I am missing something, but it seems like `action` at line 101 is set-up to include the action that actually updates the project entry in the DB. Is that right? And then, are we not updating it again at line 103?

File: app/controllers/PlutoCommissionController.scala [hunk 2]
```diff
@@ -2,25 +2,24 @@ package controllers
 
 import akka.actor.ActorRef
 import auth.BearerTokenAuth
-import exceptions.{AlreadyExistsException, BadDataException}
 import helpers.AllowCORSFunctions
-import javax.inject._
```
Comment (Dala0) [inline app/controllers/PlutoCommissionController.scala:7]: Why remove this line just to add it again on line 18?

File: app/controllers/PlutoCommissionController.scala [hunk 3]
```diff
@@ -2,25 +2,24 @@ package controllers
...
 import slick.jdbc.PostgresProfile
 import slick.jdbc.PostgresProfile.api._
 import slick.lifted.TableQuery
 
+import javax.inject._
 import scala.collection.immutable.ListMap
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.util.{Failure, Success, Try}
-import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (Dala0) [inline app/controllers/PlutoCommissionController.scala:23]: Why remove this line just to add it again on line 20?

File: app/controllers/ProjectEntryController.scala [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
 import helpers.{AllowCORSFunctions, S3Helper}
 import models._
+import play.api.Configuration
 import play.api.cache.SyncCacheApi
-import play.api.{Configuration, Logger}
 import play.api.db.slick.DatabaseConfigProvider
 import play.api.http.HttpEntity
 import play.api.inject.Injector
-import play.api.libs.json.{JsError, JsResult, JsValue, Json, Writes}
+import play.api.libs.json._
```
Comment (Dala0) [inline app/controllers/ProjectEntryController.scala:14]: Why is this line here?

Have you checked the library only includes these five functions?

File: app/controllers/ProjectEntryController.scala [hunk 5]
```diff
@@ -1,39 +1,39 @@
...
 import play.api.mvc._
+import services.RabbitMqDeliverable.DeliverableEvent
 import services.RabbitMqPropagator.ChangeEvent
+import services.RabbitMqSend.FixEvent
 import services.actors.Auditor
-import services.{CreateOperation, UpdateOperation, ValidateProject}
-import services.actors.creation.{CreationMessage, GenericCreationActor}
```
Comment (Dala0) [inline app/controllers/ProjectEntryController.scala:21]: Why remove this line just to add it again on line 21?

File: app/controllers/ProjectEntryController.scala [hunk 6]
```diff
@@ -1,39 +1,39 @@
...
 import services.actors.creation.GenericCreationActor.{NewProjectRequest, ProjectCreateTransientData}
+import services.actors.creation.{CreationMessage, GenericCreationActor}
+import services.{CreateOperation, UpdateOperation}
 import slick.dbio.DBIOAction
 import slick.jdbc.PostgresProfile
-import slick.lifted.TableQuery
```
Comment (Dala0) [inline app/controllers/ProjectEntryController.scala:25]: Why remove this line just to add it again on line 26?

File: app/controllers/ProjectEntryController.scala [hunk 7]
```diff
@@ -1,39 +1,39 @@
...
 import slick.jdbc.PostgresProfile.api._
-import services.RabbitMqSend.FixEvent
+import slick.lifted.TableQuery
+
+import java.io.File
 import java.nio.file.Paths
 import java.time.ZonedDateTime
-import scala.concurrent.{Await, Future}
```
Comment (Dala0) [inline app/controllers/ProjectEntryController.scala:30]: Why remove this line just to add it again on line 34?

File: app/controllers/ProjectEntryController.scala [hunk 8]
```diff
@@ -1,39 +1,39 @@
...
+import javax.inject.{Inject, Named, Singleton}
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.util.{Failure, Success, Try}
 import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
 import scala.language.postfixOps
-import java.io.File
```
Comment (Dala0) [inline app/controllers/ProjectEntryController.scala:35]: Why remove this line just to add it again on line 28?

File: app/controllers/ProjectEntryController.scala [hunk 9]
```diff
@@ -1,39 +1,39 @@
...
-import services.RabbitMqSend.FixEvent
+import slick.lifted.TableQuery
+
+import java.io.File
 import java.nio.file.Paths
 import java.time.ZonedDateTime
-import scala.concurrent.{Await, Future}
+import javax.inject.{Inject, Named, Singleton}
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.util.{Failure, Success, Try}
```
Comment (Dala0) [inline app/controllers/ProjectEntryController.scala:32]: Why remove this line just to add it again on line 36?

File: app/controllers/ProjectEntryController.scala [hunk 10]
```diff
@@ -69,6 +69,7 @@ class ProjectEntryController @Inject() (@Named("project-creation-actor") project
   )
 
   override def dbupdate(itemId:Int, entry:ProjectEntry) :Future[Try[Int]] = {
+    logger.info(s"Updating project ID ${itemId} and status ${entry.status}")
```
Comment (Dala0) [inline app/controllers/ProjectEntryController.scala:72]: 'ID' is more correctly written as 'id.'.

File: app/services/CommissionStatusPropagator.scala [hunk 11]
```diff
@@ -1,22 +1,23 @@
 package services
 
-import java.util.UUID
```
Comment (Dala0) [inline app/services/CommissionStatusPropagator.scala:3]: Why remove this line just to add it again on line 17?

File: test/utils/BuildMyApp.scala [hunk 12]
```diff
@@ -1,21 +1,25 @@
...
 import play.api.cache.ehcache.EhCacheModule
 import play.api.db.slick.DatabaseConfigProvider
 import play.api.inject.bind
 import play.api.inject.guice.GuiceApplicationBuilder
-import play.api.libs.concurrent.AkkaGuiceSupport
 import play.api.libs.json.Json
-import services.actors.ProjectCreationActor
 import testHelpers.TestDatabase
 
-import scala.concurrent.Future
```
Comment (Dala0) [inline test/utils/BuildMyApp.scala:15]: Why remove this line just to add it again on line 16?

File: app/controllers/PlutoCommissionController.scala (hunk 2 above)
Comment (richpryce) [reply app/controllers/PlutoCommissionController.scala:7]: IDE sorting imports automatically

**Review by Dala0 — APPROVED** [review]

**Review by paulmr — COMMENTED** [review]
As someone who doesn't know this application all that well, I can't comment too much on the overall context. But overall I think this definitely looks like the right approach!
